### PR TITLE
Update LogServiceProvider to include the syslog handler

### PR DESCRIPTION
**Pull request by synbot**
### Acceptance Criteria
1. The `LogServiceProvider` will register a syslog handler for logging

### CODE REVIEW

**Comment by spruce-bruce** [comment]
:star:

File: src/Synapse/Log/LogServiceProvider.php [hunk 1]
```diff
@@ -39,6 +40,38 @@ public function register(Application $app)
...
+        });
+
+        $app->initializer('Synapse\\Log\\LoggerAwareInterface', function ($object, $app) {
+            $object->setLogger($app['log']);
+            return $object;
+        });
+    }
+
+    /**
+     * Perform extra chores on boot (none needed here)
```
Comment (paulstatezny) [inline src/Synapse/Log/LogServiceProvider.php]: > (none needed here)

That's no longer true

File: src/Synapse/Log/LogServiceProvider.php [hunk 2]
```diff
@@ -63,28 +96,20 @@ public function register(Application $app)
...
+        return $handlers;
     }
 
-    /**
-     * Perform extra chores on boot (none needed here)
-     *
-     * @param  Application $app
-     */
-    public function boot(Application $app)
+    protected function syslogHandler($ident)
```
Comment (paulstatezny) [inline src/Synapse/Log/LogServiceProvider.php]: OCD, but function names should be verbs. Can we rename all these to `getFooHandler`?

**Comment by spruce-bruce** [comment]
:star:

**Comment by paulstatezny** [comment]
:+1:

**Comment by spruce-bruce** [comment]
:one:

**Pull request merged**